encoder/encoder.go

```diff
@@ -62,6 +62,7 @@ func Unmarshal(b []byte, v any) error {
 	return decMode.Unmarshal(b, v)
 }
 
+// why it's in not test package
```

Comment:
I believe test packages cannot be imported and this is a helper function to be used outside of encoder package